src/lower.c

```diff
@@ -4344,42 +4499,67 @@ static vcode_reg_t lower_expr(tree_t expr, expr_ctx_t ctx)
 {
    PUSH_DEBUG_INFO(expr);
 
+   if (cover_enabled(cover_tags, COVER_MASK_EXPRESSION))
+      cover_push_scope(cover_tags, expr);
```

Comment:
Why do you need to create a separate scope for each expression? I think the overhead of this will be pretty large.